src/v/cloud_topics/reconciler/reconciler_probe.cc

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "cloud_topics/reconciler/reconciler_probe.h"
+
+#include "config/configuration.h"
+#include "metrics/metrics.h"
+#include "metrics/prometheus_sanitize.h"
+
+#include <seastar/core/metrics.hh>
+
+namespace cloud_topics::reconciler {
+
+void reconciler_probe::setup_metrics() {
+    namespace sm = ss::metrics;
+
+    if (config::shard_local_cfg().disable_metrics()) {
+        return;
+    }
+
+    _metrics.add_group(
+      prometheus_sanitize::metrics_name("cloud_topics:reconciler"),
+      {
+        sm::make_counter(
+          "rounds",
+          [this] { return _rounds; },
+          sm::description("Number of reconciliation rounds")),
+        sm::make_counter(
+          "objects_uploaded",
+          [this] { return _objects_uploaded; },
+          sm::description("Total objects uploaded, including orphans")),
+        sm::make_counter(
+          "bytes_reconciled",
+          [this] { return _bytes_reconciled; },
+          sm::description("Total bytes produced, including metadata")),
+        sm::make_counter(
+          "batches_reconciled",
+          [this] { return _batches_reconciled; },
+          sm::description("Total record batches reconciled")),
```

Comment:
We should think carefully about which metrics should be per-partition. The aggregations that occur in the reconciler kind of anonymize the partitions. I guess we just need to think through the situations we might be concerned about.